### DISCUSSION TITLE
Transonic-Pyccel backend ?

### DISCUSSION THREAD

**Discussion by paugier** [pyccel/pyccel#1909] [Ideas]
Hi,

I am the creator and maintainer of [Transonic](https://transonic.readthedocs.io), a "meta Python compiler" which uses in the background other Python compilers, like Pythran, Numba, Cython and JAX.

We use Transonic for few packages of the Fluiddyn project. I'm now very happy with what it allows (first of all a very easy usage of Pythran) but I didn't try to advertise this package.

However, we recently improved Transonic to add Meson and JAX supports and I think Transonic could be interesting for Pyccel.

It seems to me that both Transonic and Pyccel are now ready to be coupled, meaning that if we add a Pyccel backend to Transonic, packages using Transonic could very easily try Pyccel and benchmark the different accelerators on their specific numerical kernels (functions).

I think adding a Pyccel backend to Transonic should not be difficult but I don't want to spend time and energy on this if there is no interest on the Pyccel side. Please tell me if someone knowing or developing Pyccel could be interested and could help a bit.

**Comment by EmilyBourne** [comment]
Hi @paugier ,

This sounds like a really nice idea. I've had a quick look at your documentation and I agree that it should be relatively easy to add a Pyccel backend. If users are using type annotations then it should be as simple as mapping `@boost` to `epyccel`.

I am happy to give you a hand with this if you want. What is your preferred method of communication? I actually already have your email from the maths.calcul mailing list (on peut communiquer en français si tu préfères).
